src/impl_float_maths.rs

```diff
@@ -0,0 +1,157 @@
+        $($(#[$meta])*
+        pub fn $id(&self, rhs: $ty) -> Array<A, D> {
+            self.mapv(|v| A::$id(v, rhs))
+        })+
+    };
+}
+
+/// # Element-wise methods for Float Array
+///
+/// Element-wise math functions for any array type that contains float number.
+impl<A, S, D> ArrayBase<S, D>
+where
+    A: Float,
+    S: RawData<Elem = A> + Data,
+    D: Dimension,
+{
+    boolean_op! {
+        /// If the number is `NaN` (not a number), then `true` is returned for each element.
+        fn is_nan
+        /// Return `true` if any element is `NaN` (not a number).
+        fn is_nan_any -> is_nan
+
+        /// If the number is infinity, then `true` is returned for each element.
+        fn is_infinite
+        /// Return `true` if any element is infinity.
+        fn is_infinite_any -> is_infinite
+    }
+    unary_op! {
+        /// The largest integer less than or equal to each element.
+        fn floor
+
+        /// The smallest integer less than or equal to each element.
+        fn ceil
+
+        /// The nearest integer of each element.
+        fn round
+
+        /// The integer part of each element.
+        fn trunc
+
+        /// The fractional part of each element.
+        fn fract
+
+        /// Absolute of each element.
+        fn abs
+
+        /// Sign number of each element.
+        ///
+        /// + `1.0` for all positive numbers.
+        /// + `-1.0` for all negative numbers.
+        /// + `NaN` for all `NaN` (not a number).
+        fn signum
+
+        /// The reciprocal (inverse) of each element, `1/x`.
+        fn recip
+
+        /// Square root of each element.
+        fn sqrt
+
+        /// `e^x` of each element. (Exponential function)
+        fn exp
+
+        /// `2^x` of each element.
+        fn exp2
+
+        /// Natural logarithm of each element.
+        fn ln
+
+        /// Base 2 logarithm of each element.
+        fn log2
+
+        /// Base 10 logarithm of each element.
+        fn log10
+
+        /// Cubic root of each element.
+        fn cbrt
+
+        /// Sine of each element. (in radians)
+        fn sin
+
+        /// Cosine of each element. (in radians)
+        fn cos
+
+        /// Tangent of each element. (in radians)
+        fn tan
+
+        /// Converts radians to degrees for each element.
+        fn to_degrees
+
+        /// Converts degrees to radians for each element.
+        fn to_radians
+    }
+    binary_op! {
+        /// Integer power of each element.
+        ///
+        /// This function is generally faster than using float power.
+        fn powi(i32)
+
+        /// Float power of each element.
+        fn powf(A)
+
+        /// Logarithm of each element with respect to an arbitrary base.
+        fn log(A)
+
+        /// The positive difference between given number and each element.
+        fn abs_sub(A)
+    }
+
+    /// Square of each element.
+    pub fn square(&self) -> Array<A, D> {
+        self.mapv(|v| v * v)
```

Comment:
I'm not sure about multiplication in Rust is faster than integer powers. Is there any reference?

Reply:
Fwiw, I agree that multiplication is probably the better implementation. I'd probably call it `pow2` instead of `square`, to minimize confusion with "square" arrays.

Reply:
Why should we have a method for squaring each element? I understand the motivation of this PR but this seems a bit too useless. It's easy enough to call mapv in this case? There is no .square() on f32, so no precedent there. :slightly_smiling_face:

Reply:
Square/sqrt is so common in scientific/statistical computing that I would like to have an implementation that does so efficiently and concisely. `powi(2)` and `powf(0.5)` are pleasantly concise, but if they are not efficient (and it looks like the consensus is that they aren't), then I would prefer an implementation that encourages users to produce faster code as is being done here.